### ISSUE TITLE
parser: JSX parsing fails depending on the `preserveParens` option

### ISSUE THREAD

**Issue by leaysgur** [oxc-project/oxc#10578]
```jsx
// conformance/jsx/jsxReactTestSuite.tsx
<Comp {...(z = { a: "1" }, z)} />
```

Our parser can parse this only when `preserveParens: true`.

If set `preserveParens: false`(for TS-ESTree), it fails with `TS(18007): JSX expressions may not use the comma operator` error.

https://playground.oxc.rs/#eNpFj0FPwzAMhf9K5BNIVWFIXCrgAtoVBNddvNSdgtK4stOxrep/x20pOyWW/d733gAeKnh65bZzQ1mWNxf37AaHldvBZgduLNzldnR3L1AAQzWA9Gl69JwynqDK0lMBMaQMVYNRbVDPHa0bPbd7juuUBZM2LO3f8VhAh6IkkyXGyD+flHtJ733WUNO2Tz4HNuAi74Ts9kgfKJT0CuRePG1DpIStkSGT5vJbT2D+U7TZ3/7/+AWYUQ5kuYH04X7zaA2DcsRM9Rv5iIITe8WY3HNNB5rrG2kf6WsGt9gtAe2kDSk0YeV5Tlk4bq3XJDqS7Fkt4GI4jr+XUoTv

---

https://github.com/oxc-project/oxc/blob/9a0206693ad0533596610c4340bd6779a3f4412f/crates/oxc_parser/src/jsx/mod.rs#L308-L310

Originally, this check is needed for code like:

```jsx
// conformance/jsx/jsxParsingError1.tsx
<div className={class1, class2}/>
```

- TS emits error for this but can parse
- Other parsers like `@typescript-eslint`, `acorn(-jsx)` also can parse and not emit error
- OXC cannot parse and emits error

I'm not sure:

- Whether it is OK to just skip this check
- How to handle correctly this without parens
